frontend/packages/knative-plugin/src/components/add/event-sources/ContainerSourceSection.tsx

```diff
@@ -0,0 +1,82 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { useFormikContext, FormikValues } from 'formik';
+import { TextInputTypes, FormGroup } from '@patternfly/react-core';
+import { InputField, MultiColumnField } from '@console/shared';
+import { AsyncComponent } from '@console/internal/components/utils';
+import FormSection from '@console/dev-console/src/components/import/section/FormSection';
+import { getSuggestedName } from '@console/dev-console/src/utils/imagestream-utils';
+
+const ContainerSourceSection: React.FC = () => {
+  const envPath = 'data.containersource.containers[0].env';
```

Comment:
If there is a case one of the values prop is undefined then please use optional chaining to get env values in a const something like `envValues`.

Reply:
I use this `envPath` in setFieldValue function below, so it will not work if i use optional chaining path in Formik and moreover the values has a default env values as initialValues which will be always present.

Reply:
Based on your previous comment, if we are destructuring then we no longer need this Path variable.

Reply:
Yes agreed! After destructuring, this is not needed. this was just a suggestion in case some values are optional.